### PR TITLE
Lock down public Service Bus endpoints

### PR DESCRIPTION
**Pull request by tompaana**
# PR for issue #441 

## What is being addressed and how

- Network rule added to Service Bus to block public endpoints

### CODE REVIEW

**Review by TessFerrandez — APPROVED** [review]

**Review by marrobi — CHANGES_REQUESTED** [review]
@tompaana I'm on leave today (public holiday in the UK) but wanted to comment before merged.

Why are we suddenly using service endpoints when we have private link configured?

Happy to chat tomorrow.

**Comment by tompaana** [comment]
> Why are we suddenly using service endpoints when we have private link configured?
> 
> Happy to chat tomorrow.

@marrobi I couldn't get it working without the service endpoints - most likely due to my lack of expertise when it comes to Terraform. Let's keep this PR open until you get back and let's have that chat tomorrow.

**Comment by tompaana** [comment]
Service Bus Networking blade after the change:
image

Access to the Service Bus from the outside is blocked:
```plaintext
Failed to send a Service Bus message: ServiceBusAuthorizationError: Ip has been prevented to connect to the endpoint.
```

File: templates/core/terraform/servicebus/variables.tf [hunk 1]
```diff
@@ -1,6 +1,6 @@
+variable "tenant_id" {}
```
Comment (marrobi) [inline templates/core/terraform/servicebus/variables.tf:1]: Is `tenand_id` and `resource_processor_subnet_id` used anywhere?

Comment (tompaana) [reply templates/core/terraform/servicebus/variables.tf]: `tenant_id` is not and is now removed.

`resource_processor_subnet_id` is used by the *Service Bus* private endpoint - which is a little strange:

```json
resource "azurerm_private_endpoint" "sbpe" {
  ...
  subnet_id           = var.resource_processor_subnet_id
```

**Review by marrobi — APPROVED** [review]
Looking good now.

**Pull request merged**